conan/cli/commands/inspect.py

```diff
@@ -1,30 +1,33 @@
 import json
 import inspect as python_inspect
+import sys
 
 from conan.api.output import ConanOutput
-from conan.cli.command import conan_command, COMMAND_GROUPS, conan_subcommand
+from conan.cli.command import conan_command, COMMAND_GROUPS
+from conan.cli.commands import default_json_formatter
 
 
-def _inspect_json_formatter(data):
-    return json.dumps(data, indent=4)
+def inspect_text_formatter(data):
+    out = ConanOutput(stream=sys.stdout)
+    for name, value in data.items():
+        if value is None:
+            continue
+        if isinstance(value, dict):
+            out.writeln(f"{name}:")
+            for k, v in value.items():
+                out.writeln(f"    {k}: {v}")
+        else:
+            out.writeln("{}: {}".format(name, value))
 
 
-@conan_command(group=COMMAND_GROUPS['consumer'])
+@conan_command(group=COMMAND_GROUPS['consumer'], formatters={"text": inspect_text_formatter, "json": default_json_formatter})
```

Comment:
The changes in this file are not all related to where the output goes. I removed the `path` subcommand from `conan inspect` because I think it was probably a mistake when porting the command? You had to invoke conan inspect like `conan inspect path conanfile.py` instead of `conan inspect conanfile.py`

Reply:
It was not a mistake, we were discussing this ``path`` as an approach to leave the possibility open for the future to inspect references in the cache and in remotes

Reply:
Maybe we should consider using a syntax more aligned with other commands, something like:
```
conan inspect conanfile.py
conan inspect --reference=zlib/1.2.11
```
I would leave the inspect on remote case for users to implement their own using custom commands.
I'll revert the changes and maybe we can discuss a change of syntax in other PR or issue.